bambi/plots/effects.py

```diff
@@ -0,0 +1,349 @@
+        the model.
+    covariates : list or dict
+        A sequence of between one and three names of variables or a dict of length between one
+        and three.
+        If a sequence, the first variable is taken as the main variable,
+        mapped to the horizontal axis. If present, the second name is a coloring/grouping variable,
+        and the third is mapped to different plot panels.
+        If a dictionary, keys must be taken from ("horizontal", "color", "panel") and the values
+        are the names of the variables.
+    target : str
+        Which model parameter to plot. Defaults to 'mean'. Passing a parameter into target only
+        works when pps is False as the target may not be available in the posterior predictive
+        distribution.
+    pps: bool, optional
+        Whether to plot the posterior predictive samples. Defaults to ``False``.
+    use_hdi : bool, optional
+        Whether to compute the highest density interval (defaults to True) or the quantiles.
+    hdi_prob : float, optional
+        The probability for the credibility intervals. Must be between 0 and 1. Defaults to 0.94.
+        Changing the global variable ``az.rcParam["stats.hdi_prob"]`` affects this default.
+    transforms : dict, optional
+        Transformations that are applied to each of the variables being plotted. The keys are the
+        name of the variables, and the values are functions to be applied. Defaults to ``None``.
+
+    Returns
+    -------
+    cap_data : pandas.DataFrame
+        A DataFrame with the ``create_cap_data`` and model predictions.
+
+    Raises
+    ------
+    ValueError
+        If passed ``covariates`` is not in correct key, value format.
+        If length of ``covariates`` is not between 1 and 3.
+
+    """
+
+    covariate_kinds = ("horizontal", "color", "panel")
+    if not isinstance(covariates, dict):
+        covariates = listify(covariates)
+        covariates = dict(zip(covariate_kinds, covariates))
+    else:
+        assert covariate_kinds[0] in covariates
+        assert set(covariates).issubset(set(covariate_kinds))
+
+    assert 1 <= len(covariates) <= 3
+
+    if transforms is None:
+        transforms = {}
+
+    if hdi_prob is None:
+        hdi_prob = az.rcParams["stats.hdi_prob"]
+
+    if not 0 < hdi_prob < 1:
+        raise ValueError(f"'hdi_prob' must be greater than 0 and smaller than 1. It is {hdi_prob}.")
+
+    cap_data = create_cap_data(model, covariates)
+    response_name = get_aliased_name(model.response_component.response_term)
+    response_transform = transforms.get(response_name, identity)
+
+    if pps:
+        idata = model.predict(idata, data=cap_data, inplace=False, kind="pps")
+        y_hat = response_transform(idata.posterior_predictive[response_name])
+        y_hat_mean = y_hat.mean(("chain", "draw"))
+    else:
+        idata = model.predict(idata, data=cap_data, inplace=False)
+        y_hat = response_transform(idata.posterior[f"{response_name}_{target}"])
```

Comment:
```python
component  = model.components[target]
if component.alias: # not sure if this is the right syntax
    component_name = get_aliased_name(component)
    idata.posterior[component_name]
else:
    idata.posterior[f"{response_name}_{target}"]
```

Reply:
I'm not sure I know what "non-parent parameter" means. Is this referring to if there's an alias on the response, then the aliased name will not be the same as the `response_name`? 

If `target != "mean"`, then logic needs to be put in place to parse the aliased name if aliased is true. Else, use the response name and default `targt="mean"`?

Reply:
```python
formula = bmb.Formula("y ~ x", "sigma ~ x")
model = bmb.Model(formula)
# y_sigma
# f"{aliased_response_name}_{parameter_name}"

# However, if you assign an alias to "sigma", then that pattern does not work anymore

aliases = {"sigma": "whatever"}

# f"{aliased_response_name}_{parameter_name}" does not work anymore
# Bambi does not prepend the response name to non-parent parameters that have an alias
# it will be 'whatever', not 'y_whatever'
```